### ISSUE TITLE
Reduce complexity of `Parameters()` func

### ISSUE THREAD

**Issue by mboersma** [kubernetes-sigs/cluster-api-provider-azure#3810]
/kind bug

**What steps did you take and what happened:**

The [`ManagedClusterSpec.Parameters()`](https://github.com/pluralsh/cluster-api-provider-azure/blob/64a02e09172a512c0f8630907be15f55ca49e0da/azure/services/managedclusters/spec.go#L270) func is long, with many `if`s and `nil` checks, so it currently exceeds the maximum cyclomatic complexity allowed by the `gocyclo` linter. We've disabled that linter for this func.

**What did you expect to happen:**

The func could be refactored to reduce [cyclomatic complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity), perhaps by breaking it into multiple funcs. Existing unit test coverage should be preserved (or improved) across the refactor, and the `//nolint:gocyclo` comment should be removed.

**Anything else you would like to add:**

You can test locally that your refactoring succeeds with `make lint`.

**Environment:**

- cluster-api-provider-azure version: `main`
- Kubernetes version: (use `kubectl version`): N/A
- OS (e.g. from `/etc/os-release`): N/A

**Comment by mboersma** [comment]
@lakshkeswani can we give you any help with this task?

**Comment by lakshkeswani** [comment]
@mboersma No Thanks, I have just started to working on it.

**Comment by souhardya79** [comment]
I  would be happy to work on it... @mboersma

**Comment by mboersma** [comment]
@souhardya79 please do, I assigned you to the task and I'm happy to review any PR.

**Comment by naveenv-devops** [comment]
Hi @mboersma, I'm Naveen — a DevOps engineer with about 5 years of experience in cluster operations and infra tooling. I'd like to take this on.

I saw @bravebeaver's earlier attempt in #4635 (closed/rotten, never merged) — I'll take a look at that approach for context before starting fresh, and make sure to preserve/improve unit test coverage and clear the //nolint:gocyclo comment as you outlined. I'll run make lint locally before opening a PR. Let me know if this is still up for grabs.

**Comment by mboersma** [comment]
/assign @naveenv-devops 

@naveenv-devops go for it, and thank you for looking at this.

**Comment by naveenv-devops** [comment]
Thanks for the assignment, @mboersma! Looking into this, it turns out `Parameters()` has already been refactored — PR #5984 (merged 2025-12-04) split it into ~20 focused helper methods and removed the `//nolint:gocyclo` directive.

I verified against current main:

- No `//nolint:gocyclo` comment exists anywhere in the repo.
- `gocyclo` isn't even in the enabled linters list in` .golangci.yml`.
- Ran the project's actual lint setup (custom `golangci-lint` binary with the `kube-api-linter` plugin, per the Makefile) against `azure/services/managedclusters/...` — 0 issues.
- Existing unit test coverage in `spec_test.go` is intact.

Looks like this can be closed. 

/close
